fix(tpc): a type annotation is necessary

Upgrading to v6.0.2 I get the following errors in the Nuxt Turnstile module:

```
.nuxt/nuxt-scripts/tpc/google-tag-manager.ts(13,17): error TS2742: The inferred type of 'useScriptGoogleTagManager' cannot be named without a reference to '.pnpm/@unhead+vue@1.9.15_vue@3.4.31_typescript@5.5.3_/node_modules/@unhead/vue'. This is likely not portable. A type annotation is necessary.
.nuxt/nuxt-scripts/tpc/google-tag-manager.ts(6,7): error TS2742: The inferred type of 'OptionSchema' cannot be named without a reference to '.pnpm/valibot@0.36.0/node_modules/valibot'. This is likely not portable. A type annotation is necessary.
.nuxt/nuxt-scripts/tpc/google-tag-manager.ts(6,7): error TS2742: The inferred type of 'OptionSchema' cannot be named without a reference to '.pnpm/valibot@0.36.0/node_modules/valibot'. This is likely not portable. A type annotation is necessary.
.nuxt/nuxt-scripts/tpc/google-analytics.ts(13,17): error TS2742: The inferred type of 'useScriptGoogleAnalytics' cannot be named without a reference to '.pnpm/@unhead+vue@1.9.15_vue@3.4.31_typescript@5.5.3_/node_modules/@unhead/vue'. This is likely not portable. A type annotation is necessary.
.nuxt/nuxt-scripts/tpc/google-analytics.ts(6,7): error TS2742: The inferred type of 'OptionSchema' cannot be named without a reference to '.pnpm/valibot@0.36.0/node_modules/valibot'. This is likely not portable. A type annotation is necessary.
.nuxt/nuxt-scripts/tpc/google-analytics.ts(6,7): error TS2742: The inferred type of 'OptionSchema' cannot be named without a reference to '.pnpm/valibot@0.36.0/node_modules/valibot'. This is likely not portable. A type annotation is necessary.
```

https://github.com/nuxt-modules/turnstile/actions/runs/9820960416/job/27116095724?pr=321#step:14:24

--- a/src/tpc/utils.ts
+++ b/src/tpc/utils.ts
@@ -37,6 +37,7 @@ export function getTpcScriptContent(input: ScriptContentOpts) {
     'import { withQuery } from \'ufo\'',
     'import { useRegistryScript } from \'#nuxt-scripts-utils\'',
     'import type { RegistryScriptInput } from \'#nuxt-scripts\'',
+    'import type { VueScriptInstance } from \'@unhead/vue\'',
   ])
 
   const chunks: string[] = []
@@ -84,7 +85,9 @@ declare global {
   chunks.push(`export type Input = RegistryScriptInput${hasParams ? '<typeof OptionSchema>' : ''}`)
 
   chunks.push(`
-export function ${input.scriptFunctionName}<T extends ${input.tpcTypeImport}>(_options?: Input) {
+export function ${input.scriptFunctionName}<T extends ${input.tpcTypeImport}>(_options?: Input): T & {
+  $script: Promise<T> & VueScriptInstance<T>;
+} {
 ${functionBody.join('\n')}
   return useRegistryScript${hasParams ? '<T, typeof OptionSchema>' : ''}('${input.tpcKey}', options => ({
         scriptInput: {

